aws/rust-runtime/aws-inlineable/src/s3_expires_interceptor.rs

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_runtime_api::box_error::BoxError;
+use aws_smithy_runtime_api::client::interceptors::context::BeforeDeserializationInterceptorContextMut;
+use aws_smithy_runtime_api::client::interceptors::Intercept;
+use aws_smithy_runtime_api::client::runtime_components::RuntimeComponents;
+use aws_smithy_types::config_bag::ConfigBag;
+use aws_smithy_types::date_time::{DateTime, Format};
+
+/// An interceptor to implement custom parsing logic for S3's `Expires` header. This
+/// intercaptor copies the value of the `Expires` header to a (synthetically added)
+/// `ExpiresString` header. It also attempts to parse the header as an `HttpDate`, if
+/// that parsing fails the header is removed so the `Expires` field in the final output
+/// will be `None`.
+#[derive(Debug)]
+pub(crate) struct S3ExpiresInterceptor;
+const EXPIRES: &str = "Expires";
+const EXPIRES_STRING: &str = "ExpiresString";
+
+impl Intercept for S3ExpiresInterceptor {
+    fn name(&self) -> &'static str {
+        "S3ExpiresInterceptor"
+    }
+
+    fn modify_before_deserialization(
+        &self,
+        context: &mut BeforeDeserializationInterceptorContextMut<'_>,
+        _: &RuntimeComponents,
+        _: &mut ConfigBag,
+    ) -> Result<(), BoxError> {
+        let headers = context.response_mut().headers_mut();
+
+        if headers.contains_key(EXPIRES) {
+            let expires_header = headers.get(EXPIRES).unwrap().to_string();
+
+            //If the Expires header fails to parse to an HttpDate we remove the header so
```

Comment:
**style** space before comment starts `// If the expires header...`

Generally see this in the codebase and would like to keep it consistent, would be nice if rustfmt did this..

Reply:
Seems like it actually used to insert that space many years ago: https://github.com/rust-lang/rustfmt/issues/1224

Reply:
Added the spaces in https://github.com/smithy-lang/smithy-rs/pull/3730/commits/6f9029d63344301677ebb7c2e15ea35f5944d667